src/main/kotlin/org/opensearch/replication/metadata/store/ReplicationMetadataStore.kt

```diff
@@ -256,4 +270,15 @@ class ReplicationMetadataStore constructor(val client: Client, val clusterServic
 
     override fun doClose() {
     }
+
+    private suspend fun checkAndWaitForStoreHealth() {
+        if(!configStoreExists()) {
+            return
+        }
+        val clusterHealthReq = ClusterHealthRequest(REPLICATION_CONFIG_SYSTEM_INDEX).waitForYellowStatus()
+        // This should ensure that security plugin and shards are active during boot-up before triggering the requests
+        val clusterHealthRes = client.suspendExecuteWithRetries(null, ClusterHealthAction.INSTANCE,
+                clusterHealthReq, log=log, defaultContext = true)
+        assert(clusterHealthRes.status <= ClusterHealthStatus.YELLOW) { "Replication metadata store is unhealthy" }
```

Comment:
what will be the behavior when it doesn't come up yellow/healthy ?

Reply:
After retries, if the store is still unhealthy, we want to surface this information and fail the request

Reply:
would throwing exception be a better behavior here ?

Reply:
Actually, the suspend call for cluster health status should either succeed or throw the exception. Any exception thrown should be handled by the caller.

Reply:
In that case  assert is not adding value and we can remove it safely?